File: test/benchmark/fib_tail_call.lua
function fib(n, a, b)
  if n == 0 then return a end
  if n == 1 then return b end
  return fib(n - 1, b, a + b)
end

local start = os.clock()
local n = 60
io.write(fib(n, 0, 1) .. "\n")
for i = 1, 400000 do
  fib(n, 0, 1)
end
io.write(string.format("elapsed: %.8f\n", os.clock() - start))

File: test/benchmark/fib_tail_call.py
from __future__ import print_function

import time

def fib(n, a, b):
  if n == 0: return a
  if n == 1: return b
  return fib(n - 1, b, a + b)

start = time.process_time()
times = 60
print(fib(times, 0, 1))

for i in range(0, 400000):
  fib(times, 0, 1)

print("elapsed: " + str(time.process_time() - start))